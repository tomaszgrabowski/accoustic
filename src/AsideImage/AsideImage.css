aside {
    flex: 1;
}

img {
    object-fit: cover;
    max-width: 100%;
}

figcaption {
    text-align: center;
    font-style: italic;
    font-size: smaller;
}
