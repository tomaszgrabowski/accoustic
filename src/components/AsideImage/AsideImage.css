aside {
    flex: 1;
}

img {
    object-fit: cover;
    max-width: 100%;
    background: url('../../img/preloader.gif') no-repeat center;
}

figcaption {
    text-align: center;
    font-style: italic;
    font-size: smaller;
}
