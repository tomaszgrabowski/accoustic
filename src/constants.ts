export const APP_NAME = 'Accoustic Home Work';
export const BASE_URL = 'https://my12.content-cms.com';
export const DATA_FETCH_URL = `${ BASE_URL }/api/859f2008-a40a-4b92-afd0-24bb44d10124/delivery/v1/content/`;
export const DEFAULT_ARTICLE_ID = 'fa9519d5-0363-4b8d-8e1f-627d802c08a8';

export const ROUTES = {
    HOME: '/',
    ABOUT_ME: '/aboutme',
    ARTICLE: '/article'
};
