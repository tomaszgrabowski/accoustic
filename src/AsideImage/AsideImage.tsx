import React from 'react';
import { BASE_URL } from '../constants';
import './AsideImage.css';

interface IProps {
    url: string;
    caption: string;
}

const AsideImage = ( props: IProps ) => {
    return (
        <aside>
            <figure>
                <img src={ `${ BASE_URL }${ props.url }` }
                     alt={ props.caption }/>
                <figcaption>{ props.caption }</figcaption>
            </figure>
        </aside>
    );
};

export default AsideImage;
