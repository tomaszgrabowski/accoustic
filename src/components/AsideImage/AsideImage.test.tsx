import { shallow } from 'enzyme';
import * as faker from 'faker';
import React from 'react';
import renderer from 'react-test-renderer';
import { BASE_URL } from '../../constants';
import AsideImage from './AsideImage';

describe( 'AsideImage', () => {
    
    test( 'given url and caption when render then show img with proper atl and caption', () => {
        const caption = faker.commerce.productName();
        const url = faker.image.nature();
        const wrapper = shallow( <AsideImage url={ url } caption={ caption }/> );
        expect( wrapper.exists( 'aside' ) ).toBeTruthy();
        expect( wrapper.find( 'img' ).prop( 'src' ) ).toEqual( `${ BASE_URL }${ url }` );
        expect( wrapper.find( 'figcaption' ).text() ).toEqual( caption );
    } );
    
    test( 'given AsideImage component when render then renders properly', () => {
        const wrapper = renderer.create( <AsideImage url={ 'url' } caption={ 'caption' }/> ).toJSON();
        expect( wrapper ).toMatchSnapshot();
    } );
} );
