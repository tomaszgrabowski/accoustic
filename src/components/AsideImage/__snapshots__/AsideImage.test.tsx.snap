// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideImage given AsideImage component when render then renders properly 1`] = `
<aside>
  <figure>
    <img
      alt="caption"
      src="https://my12.content-cms.comurl"
    />
    <figcaption>
      caption
    </figcaption>
  </figure>
</aside>
`;
